lib/grpnames.gd

```diff
@@ -16,6 +16,26 @@
 
 #############################################################################
 ##
+#O  IsTrivialNormalIntersection( <G>, <U>, <V> ) . . . . . . . generic method
+##
+##  <ManSection>
+##  <Oper Name="IsTrivialNormalIntersection" Arg="G, U, V"/>
+##
+##  <Description>
+##    For normal subgroups U and V of G, IsTrivialNormalIntersection returns
+##    true if U and V intersect trivially, and false otherwise.
+##    If U and V are not normal subgroups of G, then it may return true
+##    even if they have a nontrivial intersection.
+##  </Description>
+##  </ManSection>
+##
+DeclareOperation( "IsTrivialNormalIntersection",[IsGroup, IsGroup, IsGroup]);
+# checks if arg2 and arg3 both contain the first generator of any element in
+# the given list arg1 of (minimal) normal subgroups
+DeclareOperation( "IsTrivialNormalIntersectionInList",[IsList, IsGroup, IsGroup]);
+
```

Comment:
There is no proper documentation at all, yet. I am not sure if I really want end users to use this function. In any case, if the arguments are [L, U, V], then the algorithm will check if any of L is contained in both U and V. (Actually it only checks whether a nontrivial generator is in them.) One should use this operation (or should it be function?) when knows a priori that the intersection of U and V should contain an element of L unless the intersection is trivial. The typical example is MinimalNormalSubgroups, but e.g. DirectFactorsOfGroupFromList uses this when calling itself on a direct factor. 

In any case, I am planning to document it after reading the proper material, and it probably will not really be useful for end users. Should it be something else than an operation?